website/profile/views.py

```diff
@@ -57,15 +56,14 @@ def get_public_projects(uid=None, user=None):
 def get_public_components(uid=None, user=None):
     user = user or User.load(uid)
     # TODO: This should use User.visible_contributor_to?
-
+    # TODO: Should this be limited for users with many projects / components?
```

Comment:
Also, protip: If I had to guess, I'd say the single best performance optimization we could make for profiles would be to use a datasource with pagination, so it doesn't have to load all that DB data at once. ;)